arch/arm/core/aarch64/isr_wrapper.S

```diff
@@ -32,6 +32,7 @@ GTEXT(_isr_wrapper)
 SECTION_FUNC(TEXT, _isr_wrapper)
 	z_arm64_enter_exc x0, x1
 
+	bl	z_arm64_curr_cpu
```

Comment:
I not think this is a good idea, because z_arm64_curr_cpu will return x0, however inc_nest_counter use xreg0/xreg1.

Reply:
`xreg0/xreg1` are passed to the macro in case we want to avoid clobbering some registers (for example if we want to avoid to clobber `x0` in the macro, we pass `xreg0==x1` and `xreg1==x2` to the macro), not to actual pass parameters to the macro. Also `z_arm64_curr_cpu` belongs to the macro, doesn't make any sense to have it outside.

So, feel free to modify the macro, maybe removing the parameters if you want but move the `z_arm64_curr_cpu` inside.

Reply:
@MrVan probably I'm missing something but what's wrong with something like https://pastebin.com/RZkFXk2R ?

Reply:
ok, i though to not use fixed registers in macros. I'll follow your suggestion.

Reply:
if you do not want to use fixed registers in macro you can keep the macro as is but saving / restoring `x0` when entering / exiting the macro.